include/osquery/killswitch.h

```diff
@@ -0,0 +1,49 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#pragma once
+
+#include <boost/core/noncopyable.hpp>
+#include <osquery/expected.h>
+#include <osquery/status.h>
+#include <string>
+
+namespace osquery {
+
+class Killswitch : private boost::noncopyable {
+ public:
+  enum class SwitchOnError {
+    CallFailed = 1,
+    IncorrectResponseFormat = 2,
+    IncorrectValue = 3
+  };
+
+ private:
+  Killswitch();
+
+ public:
+  ~Killswitch();
+
+  static Killswitch& get() {
+    static Killswitch killswitch;
+    return killswitch;
+  }
+
+  Status refresh();
+
+ private:
+  Expected<bool, SwitchOnError> isSwitchOn(const std::string& key);
```

Comment:
There is one more question about interface. If I am user of this feature why should I care about errors in `Killswitch` infrastructure? If something goes wrong only thing I can do  just do not run code under switcher and noting more. As a client I can do nothing with error in 'Killswitch`. I could not event escalate it honestly. So, why should I care?

Reply:
I'm reworking these things now. You should not.
Nice catch.